test/apps/iodemo/io_demo.cc

```diff
@@ -872,6 +959,30 @@ class DemoClient : public P2pDemoCommon {
         return data_size;
     }
 
+    size_t do_io_write_am(server_info_t& server_info, uint32_t sn) {
+        size_t data_size = get_data_size();
+        bool validate    = opts().validate;
+
+        IoMessage *m = _io_msg_pool.get();
+        m->init(IO_WRITE, sn, data_size, validate);
+
+        ++server_info.num_sent;
+        ++_num_sent;
+
+        BufferIov *iov = _data_buffers_pool.get();
+        iov->init(data_size, _data_chunks_pool, sn, validate);
+
+        SendCompleteCallback *cb = _send_callback_pool.get();
+        cb->init(iov, NULL, m);
+
+        VERBOSE_LOG << "sending IO_WRITE (AM) data " << iov << " size "
+                    << data_size << " sn " << sn;
+        server_info.conn->send_am(m->buffer(), opts().iomsg_size,
+                                  (*iov)[0].buffer(), (*iov)[0].size(), cb);
```

Comment:
fragmented send/recv is not supported yet? pls add TODO and/or assertion where it makes sense

Reply:
but we have to change several places to add this, it would be easer with marked code instead of debugging segfaults (in luckky case, in worse case it can be corrupted data)

Reply:
added todo